bpf/kmesh/probes/probe.h

```diff
@@ -0,0 +1,69 @@
+#ifndef __KMESH_BPF_PROBE_H__
+#define __KMESH_BPF_PROBE_H__
+
+#include "access_log.h"
+#include "metrics.h"
+
+static inline void observe_on_pre_connect(struct bpf_sock *sk)
+{
+    struct sock_storage_data *storage = NULL;
+    if (!sk)
+        return;
+
+    storage = bpf_sk_storage_get(&map_of_sock_storage, sk, 0, BPF_LOCAL_STORAGE_GET_F_CREATE);
+    if (!storage) {
+        BPF_LOG(ERR, PROBE, "pre_connect bpf_sk_storage_get failed\n");
+        return;
+    }
+
+    storage->connect_ns = bpf_ktime_get_ns();
+    return;
+}
+
+static inline void observe_on_connect_established(struct bpf_sock *sk, __u8 direction)
+{
+    struct bpf_tcp_sock *tcp_sock = NULL;
+    struct sock_storage_data *storage = NULL;
+    __u64 flags = (direction == EGRESS) ? 0 : BPF_LOCAL_STORAGE_GET_F_CREATE;
```

Comment:
what is it for?

Reply:
Cannot we always use `BPF_LOCAL_STORAGE_GET_F_CREATE `

Reply:
This is a user-defined data area associated with sock. You need to set BPF_LOCAL_STORAGE_GET_F_CREATE flag when creating the data area for the first time.

Reply:
I think `BPF_LOCAL_STORAGE_GET_F_CREATE ` can be always set even for reading the data